CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift

```diff
@@ -0,0 +1,266 @@
+            willEndDraggingTriggered = true
+        }
+
+        func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
+            didDraggingTriggered = true
+        }
+
+        func scrollViewWillBeginDecelerating(_ scrollView: UIScrollView) {
+            willDeceleratingTriggered = true
+        }
+
+        func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
+            didDeceleratingTriggered = true
+        }
+
+        func scrollViewDidEndScrollingAnimation(_ scrollView: UIScrollView) {
+            endAnimationTriggered = true
+        }
+
+        func scrollViewDidScrollToTop(_ scrollView: UIScrollView) {
+            didScrollToTopTriggered = true
+        }
+    }
+
+    func testManager() {
+        // Given
+        let strings = ["1", "2", "3"]
+        let indexPath = IndexPath(item: 0, section: 0)
+        // When
+        let cellItems = strings.map { string in
+            TestCollectionViewCellItem(text: string)
+        }
+        viewController.manager.sectionItems = [GeneralCollectionViewSectionItem(cellItems: cellItems)]
+        wait(for: [expectation], timeout: 10)
+        //Then
+        XCTAssertEqual(viewController.collectionView.visibleCells.count, 3, "Should be displayed \(strings.count) cells")
+        for cell in viewController.collectionView.visibleCells {
+            XCTAssertTrue(cell is TestCollectionViewCell, "Cell should be an instance of `TestCollectionViewCell`")
+        }
+        let cell = viewController.collectionView.cellForItem(at: indexPath) as? TestCollectionViewCell
+        XCTAssertEqual(cell?.text, strings.first, "Cell text should be equal '\(strings[indexPath.row])'")
+    }
+
+    func testScroll() {
+        // Given
+        let numbers = Array(repeating: 0, count: 500)
+        let collectionView = viewController.collectionView
+        let delegate = TestScrollDelegate()
+        // When
+        let cellItems = numbers.map { number in
+            TestCollectionViewCellItem(text: "\(number)")
+        }
+        let sectionItem = GeneralCollectionViewSectionItem(cellItems: cellItems)
+        viewController.manager.sectionItems = [sectionItem]
+        viewController.manager.scrollDelegate = delegate
+        //Then
+        wait(for: [expectation], timeout: 10)
+        viewController.manager.scroll(to: cellItems.last!, at: .top, animated: false)
+        XCTAssertGreaterThan(viewController.collectionView.contentOffset.y, viewController.view.bounds.height)
+        let shouldScroll = viewController.collectionView.delegate?.scrollViewShouldScrollToTop?(collectionView)
+        XCTAssertTrue(shouldScroll == true)
+        XCTAssertFalse(delegate.didScrollToTopTriggered)
+        viewController.manager.scroll(to: cellItems.first!, at: .top, animated: false)
+        XCTAssertEqual(viewController.collectionView.contentOffset.y, -44)
```

Comment:
This test is failed for me.

"Test Case '-[CollectionViewToolsTests.CollectionViewToolsTests testScroll]' started.
/Users/reshetove/Documents/iOS/Projects/Rosberry/Home/CollectionViewTools/CollectionViewToolsTests/CollectionViewManager/CollectionViewManagerTests.swift:116: error: -[CollectionViewToolsTests.CollectionViewToolsTests testScroll] : XCTAssertEqual failed: ("0.0") is not equal to ("-44.0")
Test Case '-[CollectionViewToolsTests.CollectionViewToolsTests testScroll]' failed (1.035 seconds)."

Reply:
Could not reproduce